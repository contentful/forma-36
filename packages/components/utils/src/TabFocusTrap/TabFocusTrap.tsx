import React from 'react';
import { css, cx } from 'emotion';

export interface TabFocusTrapProps {
  className?: string;
  children: React.ReactNode;
}

const styles = {
  root: css({
    display: 'inherit',
    outline: 0,
  }),
};

export function TabFocusTrap({
  className,
  children,
  ...otherProps
}: TabFocusTrapProps): React.ReactElement {
  const classNames = cx(styles.root, className);

  return (
    <span tabIndex={-1} className={classNames} {...otherProps}>
      {children}
    </span>
  );
}
