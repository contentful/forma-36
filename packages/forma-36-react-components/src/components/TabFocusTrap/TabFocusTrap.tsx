import React, { Component } from 'react';
import cn from 'classnames';

const styles = require('./TabFocusTrap.css');

export interface TabFocusTrapProps {
  extraClassNames?: string;
  children: React.ReactNode;
}

export class TabFocusTrap extends Component<TabFocusTrapProps> {
  render() {
    const { extraClassNames, children, ...otherProps } = this.props;

    const classNames = cn(styles.TabFocusTrap, extraClassNames);

    return (
      <span tabIndex={-1} className={classNames} {...otherProps}>
        {children}
      </span>
    );
  }
}

export default TabFocusTrap;
