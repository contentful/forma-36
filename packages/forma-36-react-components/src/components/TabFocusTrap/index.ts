export { TabFocusTrap } from './TabFocusTrap';
export type { TabFocusTrapProps } from './TabFocusTrap';
