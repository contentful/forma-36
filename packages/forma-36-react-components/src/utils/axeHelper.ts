import { configureAxe } from 'jest-axe';

const axe = configureAxe({
  rules: {
    region: { enabled: false },
  },
});

export default axe;
