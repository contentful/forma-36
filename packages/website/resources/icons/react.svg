<svg height="46" viewBox="0 0 50 46" width="50" xmlns="http://www.w3.org/2000/svg"><g opacity=".119024"><path d="m49.9288618 23c0-3.4210526-4.1361789-6.6631579-10.4776423-8.6736842 1.4634146-6.69473685.8130081-12.02105264-2.0528455-13.7263158-.6605691-.4-1.4329268-.58947368-2.2764228-.58947368v2.34736842c.4674797 0 .843496.09473684 1.1585366.27368421 1.3821138.82105263 1.9817073 3.94736842 1.5142276 7.96842105-.1117886.9894737-.2947154 2.0315789-.5182926 3.0947368-1.99187-.5052631-4.1666667-.8947368-6.4532521-1.1473684-1.3719512-1.9473684-2.7947154-3.71578945-4.2276422-5.26315787 3.3130081-3.18947369 6.4227642-4.93684211 8.5365853-4.93684211v-2.34736842c-2.7947154 0-6.453252 2.06315789-10.152439 5.64210526-3.699187-3.55789473-7.3577236-5.6-10.152439-5.6v2.34736842c2.1036585 0 5.2235772 1.73684211 8.5365853 4.90526316-1.4227642 1.54736842-2.8455284 3.30526316-4.1971544 5.25263156-2.296748.2526316-4.4715447.6421053-6.4634147 1.1578948-.2337398-1.0526316-.406504-2.0736843-.5284553-3.0526316-.4776422-4.02105265.1117887-7.14736844 1.4837399-7.97894739.304878-.18947368.7012195-.27368421 1.1686992-.27368421v-2.34736842c-.8536586 0-1.6260163.18947368-2.296748.58947368-2.85569105 1.70526316-3.49593495 7.02105263-2.0223577 13.69473684-6.32113823 2.0210526-10.43699189 5.2526316-10.43699189 8.6631579 0 3.4210526 4.13617886 6.6631579 10.47764229 8.6736842-1.46341465 6.6947369-.81300814 12.0210526 2.0528455 13.7263158.6605691.4 1.4329268.5894737 2.2865854.5894737 2.7947154 0 6.453252-2.0631579 10.152439-5.6421053 3.699187 3.5578948 7.3577236 5.6 10.152439 5.6.8536586 0 1.6260163-.1894737 2.296748-.5894737 2.8556911-1.7052631 3.495935-7.0210526 2.0223577-13.6947368 6.300813-2.0105263 10.4166667-5.2526316 10.4166667-8.6631579zm-13.2317073-7.0210526c-.3760163 1.3578947-.843496 2.7578947-1.3719512 4.1578947-.4166667-.8421053-.8536586-1.6842105-1.3313009-2.5263158-.4674796-.8421052-.9654471-1.6631579-1.4634146-2.4631579 1.4430894.2210527 2.8353659.4947369 4.1666667.831579zm-4.6544716 11.2105263c-.7926829 1.4210526-1.605691 2.768421-2.449187 4.0210526-1.5142276.1368421-3.0487805.2105263-4.5934959.2105263-1.5345528 0-3.0691057-.0736842-4.5731707-.2-.843496-1.2526315-1.6666667-2.5894737-2.4593496-4-.7723577-1.3789473-1.4735773-2.7789473-2.1138212-4.1894737.6300813-1.4105263 1.3414635-2.8210526 2.1036586-4.2.7926829-1.4210526 1.605691-2.768421 2.449187-4.0210526 1.5142276-.1368421 3.0487805-.2105263 4.5934959-.2105263 1.5345528 0 3.0691057.0736842 4.5731707.2.843496 1.2526316 1.6666667 2.5894737 2.4593496 4 .7723577 1.3789474 1.4735773 2.7789474 2.1138212 4.1894737-.6402439 1.4105263-1.3414635 2.8210526-2.1036586 4.2zm3.2825204-1.3684211c.5487804 1.4105263 1.0162601 2.8210527 1.402439 4.1894737-1.3313008.3368421-2.7337399.6210526-4.1869919.8421053.4979675-.8105263.995935-1.6421053 1.4634146-2.4947369.4674797-.8421052.9044716-1.6947368 1.3211383-2.5368421zm-10.3048781 11.231579c-.9451219-1.0105263-1.8902439-2.1368421-2.8252032-3.3684211.9146341.0421053 1.8495934.0736842 2.7947154.0736842.9552846 0 1.9004065-.0210526 2.8252033-.0736842-.9146342 1.231579-1.8597561 2.3578948-2.7947155 3.3684211zm-7.5609756-6.2c-1.4430894-.2210527-2.8353659-.4947369-4.1666667-.831579.3760163-1.3578947.843496-2.7578947 1.3719512-4.1578947.4166667.8421053.8536586 1.6842105 1.3313009 2.5263158.4776422.8421052.9654471 1.6631579 1.4634146 2.4631579zm7.5101626-21.90526318c.9451219 1.01052632 1.8902439 2.13684208 2.8252032 3.36842108-.9146341-.0421053-1.8495934-.0736842-2.7947154-.0736842-.9552846 0-1.9004065.0210526-2.8252033.0736842.9146342-1.231579 1.8597561-2.35789476 2.7947155-3.36842108zm-7.5203252 6.19999998c-.4979675.8105263-.995935 1.6421053-1.4634146 2.4947369-.4674797.8421052-.9044716 1.6842105-1.3211383 2.5263158-.5487804-1.4105264-1.0162601-2.8210527-1.402439-4.1894737 1.3313008-.3263158 2.7337399-.6105263 4.1869919-.831579zm-9.19715448 13.1789474c-3.59756098-1.5894737-5.92479675-3.6736842-5.92479675-5.3263158s2.32723577-3.7473684 5.92479675-5.3263158c.87398374-.3894737 1.82926828-.7368421 2.81504068-1.0631579.5792683 2.0631579 1.3414634 4.2105263 2.2865853 6.4105263-.9349593 2.1894737-1.6869918 4.3263158-2.2560975 6.3789474-1.0060976-.3263158-1.96138214-.6842105-2.84552848-1.0736842zm5.46747968 15.0421053c-1.3821138-.8210527-1.9817073-3.9473685-1.5142276-7.9684211.1117886-.9894737.2947154-2.0315789.5182926-3.0947368 1.99187.5052631 4.1666667.8947368 6.4532521 1.1473684 1.3719512 1.9473684 2.7947154 3.7157895 4.2276422 5.2631579-3.3130081 3.1894737-6.4227642 4.9368421-8.5365853 4.9368421-.4573171-.0105263-.843496-.1052632-1.148374-.2842105zm24.1056911-8.0210527c.4776422 4.0210527-.1117887 7.1473684-1.4837399 7.9789474-.304878.1894737-.7012195.2736842-1.1686992.2736842-2.1036585 0-5.2235772-1.7368421-8.5365853-4.9052632 1.4227642-1.5473684 2.8455284-3.3052631 4.1971544-5.2526315 2.296748-.2526316 4.4715447-.6421053 6.4634147-1.1578948.2337398 1.0631579.4166666 2.0842106.5284553 3.0631579zm3.9126016-7.0210526c-.8739838.3894737-1.8292683.7368421-2.8150407 1.0631579-.5792683-2.0631579-1.3414634-4.2105263-2.2865853-6.4105263.9349593-2.1894737 1.6869918-4.3263158 2.2560975-6.3789474 1.0060976.3263158 1.9613821.6842105 2.8556911 1.0736842 3.597561 1.5894737 5.9247967 3.6736842 5.9247967 5.3263158-.0101626 1.6526316-2.3373983 3.7473684-5.9349593 5.3263158z"/><ellipse cx="24.989837" cy="23" rx="4.644309" ry="4.810526"/></g></svg>