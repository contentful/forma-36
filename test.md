# Just a test

1
2
