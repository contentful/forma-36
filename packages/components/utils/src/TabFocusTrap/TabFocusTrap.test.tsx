import React from 'react';
import { render } from '@testing-library/react';
import { axe } from '@/scripts/test/axeHelper';

import { TabFocusTrap } from './TabFocusTrap';

describe('TabFocusTrap', () => {
  it('renders the component', () => {
    const { getByText } = render(<TabFocusTrap>TabFocusTrap</TabFocusTrap>);

    expect(getByText('TabFocusTrap')).toBeTruthy();
  });

  it('renders the component with an additional class name', () => {
    const { container } = render(
      <TabFocusTrap className="my-extra-class">TabFocusTrap</TabFocusTrap>,
    );

    expect(container.firstChild).toHaveClass('my-extra-class');
  });

  it('has no a11y issues', async () => {
    const { container } = render(<TabFocusTrap>TabFocusTrap</TabFocusTrap>);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
