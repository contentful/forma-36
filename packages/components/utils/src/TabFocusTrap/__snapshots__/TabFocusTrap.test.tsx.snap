// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
.emotion-0 {
  display: inherit;
  outline: 0;
}

<span
  class="emotion-0"
  tabindex="-1"
>
  TabFocusTrap
</span>
`;

exports[`renders the component with an additional class name 1`] = `
.emotion-0 {
  display: inherit;
  outline: 0;
}

<span
  class="emotion-0 my-extra-class"
  tabindex="-1"
>
  TabFocusTrap
</span>
`;
